### DISCUSSION TITLE
Purpose of summing over frequencies in adjoint optimization example

### DISCUSSION THREAD

**Discussion by niccolot** [NanoComp/meep#2773] [Q&A] [answered]
Hello, in tutorial 3 of adjoint optimization ([03-Filtered_Waveguide_Bend](https://nbviewer.org/github/NanoComp/meep/blob/master/python/examples/adjoint_optimization/03-Filtered_Waveguide_Bend.ipynb)) the backprop of the mapping/filtering is done by giving to `autograd` `tensor_jacobian_product` the derivatives summed by the frequencies in the `nlopt` cost function wrapper definition:

```
gradient[:] = tensor_jacobian_product(mapping, 0)(
            v, eta_i, cur_beta, np.sum(dJ_du, axis=1)
        )  # backprop
```

I wanted to ask if this reducing strategy was chosen for a particular reason instead of others (e.g. `np.mean(dJ_du, axis=1)`), thank you in advance.

**Comment by smartalecH** [comment, marked as answer]
In this case, the FOM is a linear combination of the frequency components. The individual scaling of each frequency (whether it's a mean or something arbitrary) is extracted via autograd and injected into the adjoint current sources. So all that's left is the actual sum. 

The reason we don't hardcode the sum is because there are cases where you don't want to formulate an FOM that is a linear combination of the frequencies (eg the epigraph of a minimax, as described in the meep adjoint paper).
